modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseBuilder.java

```diff
@@ -167,6 +167,16 @@ else if (!configuredContentType.startsWith(HttpHeaders.Values.APPLICATION_X_WWW_
                     warnMapPayloadButNoUrlEncodedContentType(httpResponseHeaderBuilder.getContentType());
                 }
                 httpEntity = createUrlEncodedEntity(event, (Map) payload);
+
+                if (responseStreaming == ALWAYS && supportsTransferEncoding(event))
+                {
+                    setupChunkedEncoding(httpResponseHeaderBuilder);
+                }
+                else
+                {
+                    ByteArrayHttpEntity byteArrayHttpEntity = (ByteArrayHttpEntity) httpEntity;
```

Comment:
How are you handling here the case where the map is empty and createUrlEncodedEntity returns an EmptyHttpEntity?